sdk/analytics/src/main/java/com/klaviyo/analytics/Klaviyo.kt

```diff
@@ -41,29 +41,29 @@ object Klaviyo {
     private val preInitQueue: Queue<Operation<Unit>> = LinkedList()
 
     /**
-     * Since the analytics module owns these services, it must register them.
+     * This method is provided for apps that are unable to register their API key immediately
+     * on app launch in order enable limited SDK functionality including tracking app lifecycle,
+     * automated push token collection, and handling universal tracking links.
      *
-     * This registration is a lambda invoked when the service is required, not instantiated now
-     */
-    private fun initializeServices() = Registry.apply {
-        registerOnce<ApiClient> { KlaviyoApiClient }
-        registerOnce<State> {
-            KlaviyoState().also { state ->
-                register<StateSideEffects>(StateSideEffects(state))
-            }
-        }
-    }
-
-    /**
-     * Use this method to register Klaviyo for lifecycle functions. This is necessary for
-     * apps that are not able to [initialize] Klaviyo immediately on app launch, but would like to
-     * utilize Klaviyo Forms
+     * Your API key still must be provided as early as possible for full SDK functionality!
      *
      * @param applicationContext
      */
     fun registerForLifecycleCallbacks(applicationContext: Context) = safeApply {
-        val application = applicationContext.applicationContext as? Application
-        application?.apply {
+        if (!Registry.isRegistered<Config>()) {
```

Comment:
Extending the use of `registerForLifecycleCallbacks` a little, to capture the provided context so we can use it in subsequent calls. In particular, needed it to be able to resolve a universal tracking link's destination URL and return it to the host app (or fallback to enqueueing the click track API request)